### PR TITLE
tests: moved printf_float test to unittests

### PR DESCRIPTION
**Pull request by haukepetersen**
the `tests/printf_float` test application was (i) broken, and (ii) is better covered by unittests, so this is what I did.

Broken as `snprintf(result, str_len + 1, "%.f", floating_point_value);` should be `snprintf(result, str_len + 1, "%.1f", floating_point_value);` for the expected result...

### CODE REVIEW

**Comment by OlegHahm** [comment]
I think we should split the fix and the moving. A fix should be already provided in #5383.

**Comment by haukepetersen** [comment]
I don't care. I see this PR rather as 1 remove the test, 2 include a new unittest.

File: tests/unittests/tests-printf_float/tests-printf_float.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  unittests
+ * @{
+ *
+ * @file
+ * @brief       Implementations of unit tests for the SAUL registry
```
Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: s/the SAUL registry/printing floating point numbers/

File: tests/unittests/tests-printf_float/tests-printf_float.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+#define BUFSIZE             (10)
+
+static const float in0 = 2016.0349f;
+static const float in1 = 123.4567f;
+static const float in2 = 0.0f;
+
+static void sfprintf_float(void)
+{
+    char tmp[BUFSIZE];
+    char *str = tmp;
```
Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: why not simply 'char str[BUFSIZE]'?

File: tests/unittests/tests-printf_float/tests-printf_float.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * @{
+ *
+ * @file
+ * @brief       Unit tests for printing floating point numbers
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef TESTS_PRINTF_FLOAT_H_
+#define TESTS_PRINTF_FLOAG_H_
```
Comment (gebart) [inline tests/unittests/tests-printf_float/tests-printf_float.h]: typo `FLOAG`

File: tests/unittests/tests-printf_float/tests-printf_float.c (hunk 2 above)
Comment (gebart) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: I tried it and I got the following compilation error on every `TEST_ASSERT_EQUAL_STRING`:

```
In file included from /home/jgn/work/src/riot/sys/include/embUnit/embUnit.h:47:0,
                 from /home/jgn/work/src/riot/tests/unittests/tests-printf_float/tests-printf_float.c:24:
/home/jgn/work/src/riot/tests/unittests/tests-printf_float/tests-printf_float.c: In function ‘sfprintf_float’:
/home/jgn/work/src/riot/tests/unittests/tests-printf_float/tests-printf_float.c:39:43: error: invalid initializer
     TEST_ASSERT_EQUAL_STRING("2016.0349", str);
                                           ^
/home/jgn/work/src/riot/sys/include/embUnit/AssertImpl.h:50:44: note: in definition of macro ‘TEST_ASSERT_EQUAL_STRING’
         __typeof__(actual_) ____actual__ = actual_; \
                                            ^~~~~~~
```

Comment (A-Paul) [inline tests/unittests/tests-printf_float/tests-printf_float.c]: This line is part of a modification of *embUnit* in #1971.

**Comment by kYc0o** [comment]
closed by merging #5568

**Pull request closed without merging**